src/coprocessor/dag/aggr_fn/util.rs

```diff
@@ -57,3 +57,24 @@ pub fn rewrite_exp_for_sum_avg(schema: &[FieldType], exp: &mut RpnExpression) ->
     });
     Ok(())
 }
+
+/// Rewrites the expression to insert necessary cast functions for Bit operation family functions.
+pub fn rewrite_exp_for_bit_op(schema: &[FieldType], exp: &mut RpnExpression) -> Result<()> {
+    let ret_field_type = exp.ret_field_type(schema);
+    let ret_eval_type = box_try!(EvalType::try_from(ret_field_type.tp()));
+    let new_ret_field_type = match ret_eval_type {
+        EvalType::Int => {
+            return Ok(());
+        }
+        _ => FieldTypeBuilder::new()
```

Comment:
Looks like we will meet some duplicate code here. We build a field type when it is used in schema, and build another one when it is used to cast. We can refine this in future PRs, for all aggregation executors.

Reply:
OK